src/main/java/com/google/gcs/sdrs/service/worker/impl/ExecuteRetentionWorker.java

```diff
@@ -128,8 +128,32 @@ private void executePolicyByProject(String projectId) throws IOException, SQLExc
     List<RetentionRule> datasetRules = retentionRuleDao.findDatasetRulesByProjectId(projectId);
     List<RetentionRule> defaultRules = retentionRuleDao.findDefaultRulesByProjectId(projectId);
     RetentionRule globalDefaultRule = retentionRuleDao.findGlobalRuleByProjectId(projectId);
-    if (datasetRules == null || defaultRules == null || globalDefaultRule == null) {
-      throw new SQLException("Failed to get rules");
+    boolean noDatasetRules = false;
+    boolean noDefaultRules = false;
+    boolean noGlobalRule = false;
+
+    if (datasetRules == null || datasetRules.isEmpty()) {
+      logger.info(
+          String.format("No dataset rules configured within SDRS for project %s.", projectId));
```

Comment:
do we need a guard to check if projectId is null or are we already checking that higher up in the call stack?

Reply:
we are checking at the caller.